typescript/cli/src/deploy/warp.ts

```diff
@@ -110,20 +112,30 @@ async function executeDeploy(params: DeployParams) {
     context: { registry, multiProvider, isDryRun, dryRunChain },
   } = params;
 
-  const deployer = configMap.isNft
-    ? new HypERC721Deployer(multiProvider)
-    : new HypERC20Deployer(multiProvider);
-
-  const config: WarpRouteDeployConfig =
+  const warpDeployconfig: WarpRouteDeployConfig =
     isDryRun && dryRunChain
       ? { [dryRunChain]: configMap[dryRunChain] }
       : configMap;
 
-  const deployedContracts = await deployer.deploy(config);
+  const deployedAddressesMap = await promiseObjAll(
+    objMap(warpDeployconfig, async (chain, config) => {
+      const evmERC20WarpModule = await EvmERC20WarpModule.create({
+        chain,
+        config,
+        multiProvider,
+      });
 
+      return evmERC20WarpModule.serialize();
+    }),
+  );
   logGreen('✅ Hyp token deployments complete');
 
-  const warpCoreConfig = await getWarpCoreConfig(params, deployedContracts);
+  const deployedContractMap = attachContractsMap(
```

Comment:
can you do this inside the map above? or this function expects a map vs just attaching as single one at a time

Reply:
Yup, i will keep that in mind when i re-implement this in another PR